src/components/SignUp.tsx

```diff
@@ -8,26 +8,56 @@ import { Grid } from "@mui/material";
 import Button from "@mui/material/Button";
 import Link from "@mui/material/Link";
 import TextField from "@mui/material/TextField";
-//import SuccessAlert from './SuccessAlert';
 
 function SignUp() {
-  // States
-//  const [successMsg, setMessage] = useState<string | null>(null);
+  // All the error messages are stored in state, so we can display them in the form
   const [firstNameError, setFirstNameError] = useState("");
   const [lastNameError, setLastNameError] = useState("");
   const [emailError, setEmailError] = useState("");
   const [passwordError, setPasswordError] = useState("");
-//  const [openAlert, setOpenAlert] = useState(false);
 
   // Handle submit button press
   const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     const data = new FormData(event.currentTarget);
+
     // Get form data, if not available set to empty string
-    const firstname = data.get("firstName") || "";
-    const lastname = data.get("lastName") || "";
-    const email = data.get("email") || "";
-    const password = data.get("password") || "";
+    const firstname = (data.get("firstName") || "") as string;
+    const lastname = (data.get("lastName") || "") as string;
+    const email = (data.get("email") || "") as string;
+    const password = (data.get("password") || "") as string;
+    
+    // Reset all errors before validation
+    setFirstNameError("");
+    setLastNameError("");
+    setEmailError("");
+    setPasswordError("");
+
+    // Simple form validation
+    let hasErrors = false; // shitty way of doing this, but it works
+    if (firstname.trim() === "") { // trim removes whitespace
+      setFirstNameError("First name is required");
+      hasErrors = true;
+    }
+
+    if (lastname.trim() === "") {
+      setLastNameError("Last name is required");
+      hasErrors = true;
+    }
+
+    if (!/^[\w-]+(\.[\w-]+)*@([\w-]+\.)+[a-zA-Z]{2,7}$/.test(email)) {
+      setEmailError("Please enter a valid email address");
+      hasErrors = true;
+    }
+
+    if (password.length < 6) {
+      setPasswordError("Password must be at least 6 characters long");
+      hasErrors = true;
+    }
+
+    if (hasErrors) {
+      return;
+    }
```

Comment:
You can get rid of the shitty way like this :)
```suggestion
    if (firstname.trim() === "") { // trim removes whitespace
      setFirstNameError("First name is required");
      return;
    }

    if (lastname.trim() === "") {
      setLastNameError("Last name is required");
      return;
    }

    if (!/^[\w-]+(\.[\w-]+)*@([\w-]+\.)+[a-zA-Z]{2,7}$/.test(email)) {
      setEmailError("Please enter a valid email address");
      return;
    }

    if (password.length < 6) {
      setPasswordError("Password must be at least 6 characters long");
      return;
    }
```

Reply:
Makes sense. Thanks for explaning